src/json.hpp

```diff
@@ -7589,11 +7599,27 @@ class basic_json
     7159](http://rfc7159.net/rfc7159), because any order implements the
     specified "unordered" nature of JSON objects.
     */
-    using object_t = ObjectType<StringType,
-          basic_json,
-          std::less<StringType>,
-          AllocatorType<std::pair<const StringType,
-          basic_json>>>;
+
+#if defined(JSON_HAS_STRING_VIEW)
+	// if std::string_view is to be used the object_t must
+	// be declared with a transparent comparator
+	// https://stackoverflow.com/questions/35525777/use-of-string-view-for-map-lookup
+	using object_t = ObjectType<StringType,
+		basic_json,
+		std::less<>,
+		AllocatorType<std::pair<const StringType,
+		basic_json>>>;
+
+	using object_comparator_key_t = std::string_view;
+#else
+	using object_t = ObjectType<StringType,
+		basic_json,
+		std::less<StringType>,
+		AllocatorType<std::pair<const StringType,
+		basic_json>>>;
+
+	using object_comparator_key_t = typename object_t::key_type;
+#endif
```

Comment:
Any reason not to do the transparent comparator across the board for C++14 and higher?